File: go/068-prime-factors/prime_factors.go
package prime

func Factors(n int64) []int64 {
	var factors []int64
	var divisor int64 = 2
	remaining := n

	for remaining > 1 {
		if remaining%divisor == 0 {
			factors = append(factors, divisor)
			remaining = remaining / divisor
		} else {
			divisor += 1
		}
	}

	return factors
}

File: ruby/095-prime-factors/prime_factors.rb
class PrimeFactors
  # @param[Numeric]num
  def self.of(num)
    return [] if num == 1

    factors = []
    remainder = num
    divisor = 2
    while remainder > 1
      if remainder % divisor == 0
        factors << divisor
        remainder /= divisor
      else
        divisor += 1
      end
    end

    factors
  end
end